src/functional/ffront/dialect_parser.py

```diff
@@ -61,7 +61,7 @@ class DialectParser(ast.NodeVisitor, Generic[DialectRootT]):
     captured_vars: CapturedVars
     externals_defs: dict[str, Any]
 
-    syntax_error_cls: ClassVar[Type[DialectSyntaxError]] = "DialectSyntaxError"
+    syntax_error_cls: ClassVar[Type[DialectSyntaxError]] = field(default=lambda: DialectSyntaxError)
```

Comment:
I don't think this works as expected, since class members which are `ClassVar`s are not processed by dataclasses. I don't know if I understand the intended purpose here, but if `syntax_error_cls` can be a different class in different subclasses, I think there should be another `ErrorT` TypeVar (`ErrorT = TypeVar(ErrorT, bound=DialectSyntaxError)`

and then:
```
class DialectParser(ast.NodeVisitor, Generic[RootT, ErrorT]):
    ...
    syntax_error_cls: ClassVar[Type[ErrorT]] = DialectSyntaxError
```

Does any of this make sense to you?

Reply:
It does make sense but the intention is that error types in subclasses have to be subclasses of `DialectSyntaxError`. Isn't that already expressed here?

The reason for assigning the base exception class here is that otherwise mypy complains that the dataclass constructor isn't passed the mandatory keyword arg `syntax_error_cls` somewhere down the line. I could also just silence that mypy error 🤷 .

Reply:
Yes, the typing annotation here expresses exactly what you say, but I'm confused with the `= field(...)` assignment, since a `ClassVar` should be ignored by `dataclasses` and it will be never requested at initialization.

Shouldn't be enough to move the definition of `DialectSyntaxError` in front of `DialectParser` so this works as it should:
```
syntax_error_cls: ClassVar[Type[DialectSyntaxError]] = DialectSyntaxError
```

Reply:
Yes, that is the other solution. Let's go with that.